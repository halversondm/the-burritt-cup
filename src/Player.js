import React, { Component } from 'react';
import Row from "react-bootstrap/Row";
import Col from 'react-bootstrap/Col';
import PropTypes from 'prop-types';
import Lightbox from 'react-images';

class Player extends Component {

    constructor(props) {
        super(props);
        this.state = { lightboxIsOpen: false };
        this.closeLightbox = this.closeLightbox.bind(this);
        this.openLightbox = this.openLightbox.bind(this);
    }

    closeLightbox() {
        this.setState({
            lightboxIsOpen: false
        });
    }

    openLightbox(event) {
        event.preventDefault();
        this.setState({
            lightboxIsOpen: true
        });
    }

    render() {
        return (
            <div>
                <hr />
                <Row className="justify-content-center">
                    <Col lg={8} md={10}>
                        <div className="post-preview">
                            <h2 className="post-title">
                                <div className="popup-gallery">
                                    <a href={this.props.image} onClick={this.openLightbox}>
                                        <img className="head" src={this.props.thumbnail} alt={this.props.alt} />
                                    </a> {this.props.name}
                                </div>
                            </h2>
                            {this.props.office !== "" ? <h3 className="post-subtitle">{this.props.office}</h3> : null}
                            <h4 className="post-subtitle">
                                Hometown: {this.props.hometown} <br /> Nickname: {this.props.nickname} <br />
                            </h4>
                            <p className="post-meta">{this.props.about}</p>
                        </div>
                        <Lightbox images={[{src: this.props.image}]} onClose={this.closeLightbox}
                                isOpen={this.state.lightboxIsOpen} />
                    </Col>
                </Row>
            </div>
        );
    }
}

Player.defaultProps = {
    image: "",
    thumbnail: "",
    alt: "",
    name: "",
    office: "",
    hometown: "",
    nickname: "",
    about: ""
};

Player.propTypes = {
    image: PropTypes.string,
    thumbnail: PropTypes.string,
    alt: PropTypes.string,
    name: PropTypes.string,
    office: PropTypes.string,
    hometown: PropTypes.string,
    nickname: PropTypes.string,
    about: PropTypes.string
};

export default Player;