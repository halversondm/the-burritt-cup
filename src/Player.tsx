import Row from "react-bootstrap/Row";
import Col from 'react-bootstrap/Col';

interface PlayerProps {
    image: string,
    thumbnail: string,
    alt: string,
    name: string,
    office: string,
    hometown: string,
    nickname: string,
    about: string
}

export default function Player({image, thumbnail, alt, name, office, hometown, nickname, about}: PlayerProps) {

    return (
        <div>
            <hr/>
            <Row className="justify-content-center">
                <Col lg={8} md={10}>
                    <div className="post-preview">
                        <h2 className="post-title">
                            <a href={image}>
                                <img className="head" src={thumbnail} alt={alt}/>
                            </a> {name}
                        </h2>
                        {office !== "" ? <h3 className="post-subtitle">{office}</h3> : null}
                        <h4 className="post-subtitle">
                            Hometown: {hometown} <br/> Nickname: {nickname} <br/>
                        </h4>
                        <p className="post-meta">{about}</p>
                    </div>
                </Col>
            </Row>
        </div>
    );
}