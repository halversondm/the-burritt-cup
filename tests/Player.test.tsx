import Player from '../src/Player';
import {render} from "@testing-library/react";

test("renders without crashing", () => {
  const unit = <Player />;
  const component = render(unit);
  const tree = component.container;
  expect(tree.toString()).toMatchSnapshot();
});