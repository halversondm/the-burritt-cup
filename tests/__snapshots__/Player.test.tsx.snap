// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders without crashing 1`] = `"<div><div><hr><div class="justify-content-center row"><div class="col-lg-8 col-md-10"><div class="post-preview"><h2 class="post-title"><a><img class="head"></a> </h2><h3 class="post-subtitle"></h3><h4 class="post-subtitle">Hometown:  <br> Nickname:  <br></h4><p class="post-meta"></p></div></div></div></div></div>"`;
